### PR TITLE
JSON RPC support

### PR DESCRIPTION
**Pull request by ilblackdragon**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -33,3 +33,7 @@ tmp/
 
 # Logs
 *.log
+
+# Nearlib files & coverage
+nearlib/neardev/
+nearlib/coverage/
```
Comment (janedegtiareva) [inline .gitignore]: thanks for adding these!

File: node/http/src/server.rs [hunk 2]
```diff
@@ -268,7 +271,7 @@ fn serve<T: Send + Sync + 'static>(http_api: Arc<HttpApi<T>>, req: Request<Body>
                 }
             }))
         }
-        (&Method::GET, "/healthz") => {
+        (&Method::GET, "/healthz") | (&Method::GET, "/status") => {
```
Comment (janedegtiareva) [inline node/http/src/server.rs]: is the plan to make these different in the future or remove healthz?

File: node/http/src/types.rs [hunk 3]
```diff
@@ -14,6 +14,13 @@ use primitives::types::{
     AccountId, AuthorityStake, Balance, BlockIndex, MerkleHash, Nonce, ShardId,
 };
 
+pub enum RPCError {
+    BadRequest(String),
```
Comment (janedegtiareva) [inline node/http/src/types.rs]: Do we want to document these?

Comment (janedegtiareva) [inline node/http/src/types.rs]: do we want some sort of generic "server error" in addition to these?

File: nearlib/account.js [hunk 4]
```diff
@@ -53,7 +53,7 @@ class Account {
             createAccount,
             signature,
         });
-        return await this.nearClient.submitTransaction(signedTransaction);
+        return this.nearClient.submitTransaction(signedTransaction);
```
Comment (janedegtiareva) [inline nearlib/account.js]: most other code in nearlib does an await for the async functions in nearlib. Why do it differently here?

Comment (ilblackdragon) [reply nearlib/account.js]: https://eslint.org/docs/rules/no-return-await

Comment (ilblackdragon) [reply nearlib/account.js]: Apparently we don't run eslint, even though we have the config?

File: node/http/src/types.rs (hunk 3 above)
Comment (ilblackdragon) [reply node/http/src/types.rs]: Prefer adding things when we actually need them.

File: nearlib/account.js (hunk 4 above)
Comment (vgrichina) [reply nearlib/account.js]: @ilblackdragon adding step to gitlab config

File: .gitignore [hunk 5]
```diff
@@ -33,3 +33,7 @@ tmp/
 
 # Logs
 *.log
+
+# Nearlib files & coverage
+nearlib/neardev/
```
Comment (vgrichina) [inline .gitignore]: Put this into `nearlib/.gitignore` so that we can move `nearlib` and settings still apply?

File: nearlib/nearclient.js [hunk 6]
```diff
@@ -8,39 +8,85 @@ function _arrayBufferToBase64(buffer) {
...
 
-    async getNonce (account_id) {
-        return (await this.viewAccount(account_id)).nonce + 1;
+    async getTransactionStatus(transactionHash) {
+        const encodedHash = _arrayBufferToBase64(transactionHash);
+        const response = await this.jsonRpcRequest('tx', [encodedHash, false]);
+        // tx_result has default values: code = 0, logs: '', data: ''.
+        let status;
+        const code = response.tx_result.code || 0;
+        switch (code) {
```
Comment (vgrichina) [inline nearlib/nearclient.js]: nit: just have a map/array for lookup?

File: nearlib/nearclient.js [hunk 7]
```diff
@@ -8,39 +8,85 @@ function _arrayBufferToBase64(buffer) {
...
+    async getNonce(accountId) {
+        return (await this.viewAccount(accountId)).nonce + 1;
+    }
+
+    async jsonRpcRequest(method, params) {
+        const request = {
+            jsonrpc: '2.0',
+            method,
+            params,
+            id: 'dontcare',
```
Comment (vgrichina) [inline nearlib/nearclient.js]: are you sure there's not gonna be collision of ids? Like maybe it matters when you issue multiple requests at the same time?

File: core/primitives/src/transaction.rs [hunk 8]
```diff
@@ -898,6 +898,17 @@ pub enum FinalTransactionStatus {
     Completed,
 }
 
+impl FinalTransactionStatus {
+    pub fn to_code(&self) -> u64 {
+        match self {
+            FinalTransactionStatus::Completed => 0,
+            FinalTransactionStatus::Failed => 1,
+            FinalTransactionStatus::Started => 2,
+            FinalTransactionStatus::Unknown => 9999,
```
[the remark this answers is not in the document]
Comment (ilblackdragon) [reply core/primitives/src/transaction.rs]: 0 is OK by default

File: nearlib/nearclient.js (hunk 7 above)
Comment (ilblackdragon) [reply nearlib/nearclient.js]: made it a timestamp

File: node/http/src/api.rs [hunk 9]
```diff
@@ -1,40 +1,194 @@
...
 
 use crate::types::*;
-use primitives::transaction::verify_transaction_signature;
-use primitives::transaction::SignedTransaction;
+use node_runtime::state_viewer::AccountViewCallResult;
 
 const POISONED_LOCK_ERR: &str = "The lock was poisoned.";
 
+/// Adapter for querying runtime.
+pub trait RuntimeAdapter {
```
[the remark this answers is not in the document]
Comment (ilblackdragon) [reply node/http/src/api.rs]: To re-use this logic in other places.

**Pull request merged**